Content.Server/Nutrition/EntitySystems/FlavorProfileSystem.cs

```diff
@@ -25,12 +25,12 @@ public sealed class FlavorProfileSystem : EntitySystem
     public string GetLocalizedFlavorsMessage(EntityUid uid, EntityUid user, Solution solution,
         FlavorProfileComponent? flavorProfile = null)
     {
-        var flavors = new HashSet<string>();
         if (!Resolve(uid, ref flavorProfile, false))
         {
             return Loc.GetString(BackupFlavorMessage);
         }
 
+        var flavors = new HashSet<string>();
         flavors.UnionWith(flavorProfile.Flavors);
```

Comment:
Can you pass the hashset into the ctor or is that just lists? At a minimum should initialize the capacity. to flavorProfile.Flavors.Count.